sympy/polys/polyclasses.py

```diff
@@ -2759,35 +2759,49 @@ def init_normal_ANP(rep, mod, dom):
 class ANP(PicklableWithSlots, CantSympify):
     """Dense Algebraic Number Polynomials over a field. """
 
-    __slots__ = ('rep', 'mod', 'dom')
+    __slots__ = ('_rep', '_mod', 'dom')
 
     def __init__(self, rep, mod, dom):
         # Not possible to check with isinstance
         if type(rep) is dict:
-            self.rep = dup_from_dict(rep, dom)
+            rep = DMP(dup_from_dict(rep, dom), dom, 0)
         else:
             if isinstance(rep, list):
                 rep = [dom.convert(a) for a in rep]
+            elif isinstance(rep, DMP):
+                rep = rep.to_list()
             else:
                 rep = [dom.convert(rep)]
 
-            self.rep = dup_strip(rep)
+            rep = DMP(dup_strip(rep), dom, 0)
 
         if isinstance(mod, DMP):
-            self.mod = mod.rep
+            pass
         else:
             if isinstance(mod, dict):
-                self.mod = dup_from_dict(mod, dom)
+                mod = DMP(dup_from_dict(mod, dom), dom, 0)
             else:
-                self.mod = dup_strip(mod)
+                mod = DMP(dup_strip(mod), dom, 0)
 
+        self._rep = rep
+        self._mod = mod
         self.dom = dom
 
+    @property
+    def rep(self):
+        raise RuntimeError("ANP.rep")
+        return self._rep.to_list()
+
+    @property
+    def mod(self):
+        raise RuntimeError("ANP.mod")
+        return self._mod.to_list()
```

Comment:
This always raise errors. Are you leaving this because this is part of some incomplete work?

Reply:
It is incomplete. I haven't decided whether or not to deprecate these. The `rep` attribute can be replaced with `.to_list()` just like for `DMP`. The mod argument does not have an existing replacement. It is perhaps also not too important to deprecate them but I do want to make sure that they aren't use within the sympy codebase itself.

Reply:
I think that although keeping the `rep, dom` as list is backward compatible, I don't think that it is more useful than using instance of `DMP` directly.

Reply:
I guess at least there should be new public methods/attributes to get the rep and mod as DMP.